embassy-stm32/src/dma/bdma.rs

```diff
@@ -270,10 +364,31 @@ mod low_level_api {
 
         if isr.teif(channel_num) {
             panic!("DMA: error on BDMA@{:08x} channel {}", dma.0 as u32, channel_num);
+        } else if STATE.data_ready[index].load(Ordering::SeqCst) {
+            if cr.read().dir() == vals::Dir::FROMPERIPHERAL {
+                panic!("DMA: Concurrent access on the same buffer half between CPU and BDMA@{:08x} channel {}. User data processing may be too slow", dma.0 as u32, channel_num);
+            } else {
+                panic!("DMA: Concurrent access on the same buffer half between CPU and BDMA@{:08x} channel {}.Writting on the buffer by the user may be too slow", dma.0 as u32, channel_num);
```

Comment:
This direction is not technically an UB, just random data being sent out to peripheral. Maybe it would be better to report an error instead of panic? Although, it would complicate code a bit and might be better left for future PR.

Reply:
So on your opinion, I should just delete it for now ?

Reply:
No, I think it's okay as is. It can be addressed in another PR adding different transfer direction (i.e. DAC) to keep this concise.